File: leetcode/python/539.minimum-time-difference.py
class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        minutes = [int(timePoint[:2]) * 60 + int(timePoint[3:]) for timePoint in timePoints]
        minutes.sort()
        ans = 1440

        for i in range(1, len(timePoints)):
            ans = min(ans, minutes[i]-minutes[i-1])
        
        ans = min(ans, 1440 - minutes[-1] + minutes[0])
        
        return ans

File: leetcode/cpp/539.minimum-time-difference.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        vector<int> minutes;
        for (const string& timePoint : timePoints) {
            int hour = stoi(timePoint.substr(0, 2));
            int minute = stoi(timePoint.substr(3, 2));
            minutes.push_back(hour * 60 + minute);
        }

        sort(minutes.begin(), minutes.end());

        int ans = 1440;
        for (int i = 1; i < minutes.size(); ++i) {
            ans = min(ans, minutes[i] - minutes[i - 1]);
        }

        ans = min(ans, 1440 - minutes.back() + minutes.front());

        return ans;
    }
};

